cookiecutter/generate.py

```diff
@@ -339,6 +340,10 @@ def generate_files(
                 outdir = os.path.normpath(os.path.join(project_dir, indir))
                 outdir = env.from_string(outdir).render(**context)
                 logger.debug('Copying dir %s to %s without rendering', indir, outdir)
+
+                # if the outdir is there, it must be a overwrite execution
+                if os.path.isdir(outdir):
+                    shutil.rmtree(outdir)
```

Comment:
I don't think this is the correct solution. I think #1661 is.

Reply:
As discussed in #1661. I think this is the correct solution for all versions of python.

```suggestion
                # if the outdir is there, it must be a overwrite execution
                if os.path.isdir(outdir) and overwrite_if_exists:
                    shutil.rmtree(outdir)
```